### ISSUE TITLE
Stand der Dinge bei mir

### ISSUE THREAD

**Issue by lisapillep** [timliebold/DS-Hausaufgaben#2]
Habe jetzt soweit alles installiert, mir die Funktionen mal gaaaanz grob angschaut und das Projekt runtergeladen. Vllt können wir uns ja nächste Woche mal alle online treffen, damit wir alle aufm gleichen Wissensstand sind.

**Comment by timliebold** [comment]
jopp das klingt nach nem sehr guten plan. 

Die Grundfunktion einer Versionsverwaltung wie github liegt darin aus einem einzelnen entwicklungsstrang versionen auszuklinken und diese weiter zu entwickeln um sie dann wieder mit dem hauptstrang zu vereinen. Der clou daran ist aber das man sich dadurch nicht gegenseitig dateien überschreiben kann. Wir können also tatsächlich alle gleichzeitig an den gleichen Dateien arbeiten.

hier läuft das über branches.
wir haben schon einen der sich master nennt und das ist unser hauptbranch. an dem machen wir niemals änderungen von hand. aber wir können den branch clonen und einen zweiten master branch mit anderem namen kreieren. daran können wir dann wild rumbauen. und im nachhinein schauen sich die anderen den code an  und wenn er gefällt wird er in den masterbranch gemerged und die änderung ist in die nächste version mit eingeflossen :+1:

**Comment by xRobertx** [comment]
Jo, also ich habe jetzt auch die .rar-Datei runtergeladen und git installiert. Was ich mit git mache ist mir jetzt noch unklar...
Aber ich soll meine vorhandenen .java Dateien des Projektes in die .rar hauen und die dann hier hochladen, oder wie? Also ich habe alle Klassen und Rollen implementiert. Es fehlen nur noch die Methoden/Zuordnungen aus Aufgabe 2. Da hab ich gerade n Brett vor'm Kopf und mache keine Fortschritte, deswegen würde ich euch meinen bsherigen Stand gerne mal zukommen lassen.
Dann könnt ihr auch gleich mal auf evtl. Fehler etc. durchschauen

**Comment by timliebold** [comment]
okay vielleicht schreibst du mal welchen git client du benutzt. und schickst mal zwei drei screenshots mit (geht hier einfach mittels einfügen). welche rar datei meinst du? 

wenn du das projekt clonst, erzeugst du parallel die verbindung (befehl git clone)
![image](https://cloud.githubusercontent.com/assets/19992362/25770934/ff08b5d0-3242-11e7-8a53-5b788ece156e.png)

mit git status im geclonten repo sehe ich dass da nur der master vorhanden ist (siehe bild)
dann checkst du mit git checkout -b einen neuen branch aus und nimmst da deine änderungen rein
![image](https://cloud.githubusercontent.com/assets/19992362/25770964/af1cc48e-3243-11e7-9194-4dd9195db74e.png)
bei mir ist das jetzt Beispiel.
und im Beispiel branch nehme ich jetzt änderungen vor.
wenn ich dann git status mache sehe ich auch meine änderungen und mittels git add und git commit kann ich die änderung für diesen einen branch speichern
![image](https://cloud.githubusercontent.com/assets/19992362/25770990/7ea99f74-3244-11e7-86f9-14e9ee7c9096.png)
wenn ich die änderung dann offiziell machen möchte nutze ich git push und pushe damit meinen branch hier zu github
![image](https://cloud.githubusercontent.com/assets/19992362/25771001/d5b5e4b2-3244-11e7-95a1-a9ef0cdcce5e.png)
jetzt sollt ihr den code ja noch sehen, also erstelle ich ein pull request in dem ich oben auf code drücke und dann den anweisungen folge. das war git und zwar alles ;)

**Comment by timliebold** [comment]
wenn ihr unter dem reiter code mal meinen BeispielBranch auswählt seht ihr auch welche änderung ich gemacht habe. ebenso im pullrequest

**Comment by timliebold** [comment]
und ganz unten im pull request steht merge. damit würden wir die änderung dann übernehmen und in den master mergen

**Comment by timliebold** [comment]
@xRobertx dir würde ich jetzt empfehlen das repo neu zu klonen, einen neuen branch auszuchecken und deine dateien dateien in den geclonten ordner zu kopieren. dort machst du dann git add -A und hast alle dateien zu dem branch hinzugefügt den du zuvor ausgecheckt hast. dann git commit -m '<message>' und dann git push. du bekommst mit sicherheit so wie es bei mir oben auch war noch ein paar fragen zu username und email und paswort und zu deiner push methode (ich hab sie auf matching gestellt und kann das nur weiteremfehlen) beim push wird dir dann auch noch der befehl zum tracken des upstream branches vorgeschlagen, den kannst du einfach kopieren. damit erstellst du dann hier (origin) einen neuen Branch mit dem ggleichen namen wie deinem. wenn du dann git push machst pusht git immer in den gematchten branch. dann noch pullrequest erstellen und fertig

**Comment by timliebold** [comment]
wenn ihr im übrigen bei git checkout das "-b" weglasst könnt ihr zwischen existierenden branches wechseln und git branch zeigt euch alle existierenden branches an. desweiteren könnt ihr mit git branch -D einen existierenden branch mit all seinen änderungen löschen. 
prüft eure commits und pushes auch immer nochmal mit git status. da seht ihr dann nicht gepushte commits und nicht committete änderungen ;)

**Comment by xRobertx** [comment]
![1](https://cloud.githubusercontent.com/assets/28360775/25894645/e9c04fca-357c-11e7-9b01-2a517ab5175d.PNG)

Okay, also ich habe ienfach nach git-client gegoogelt und das gefunden und installiert - sieht irgendwie komplett anders aus als bei dir...
Und die anbei stehende "DS-HA..." habe ich einfach hier heraus geladen, nicht mit dem Client geklont.

**Comment by timliebold** [comment]
ja aber die solltest du mit dem client clonen da bekommst du bissl mehr zurück

ich denke du kannst das ebenso wie ich wie ein stinknormales terminal benutzen und meine befehle nehmen 

also gehst du in den ordner wo dus gern hättest (cd, ls, ...) und dann:
1. git clone <url vom repo>
2. git branch -> da stehen jetzt alle branches die du geclont hast

wenn du jetzt was ändern magst
git checkout -b <name des neuen branches> -> erstellt einen neuen branch, den du auch gleich auscheckst
mit git branch kannst du jetzt sehen dass du den branch erstellt und ausgecheckt hast und dass er neben dem master liegt, den du mit git checkout master auch jederzeit auschecken kannst

wenn du deine änderungen gemacht hast pushst du den branch mit git push
wenn er nicht pusht machst du git push --set-upstream origin <branchname> (der branchname ist identisch zu deinem lokalen branch)

dann oben auf code drücken und PR öffnen
fertig.

wenn jetzt hier ein PR gemerged wird also in den master mit einfließt kannst du das update laden, indem du den master auscheckst und git pull machst
